File: 프로그래머스/lv1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.js
function solution(board, moves) {
    var answer = 0;

    let temp = [];
    for(let i = 0; i < board[0].length; i += 1){
        let tempArr = [];
        board.map(el => {
            if (el[i] > 0) {
                tempArr.push(el[i]);
            }
        });
        temp.push(tempArr);
    }

    let stack = [];
    for (const idx of moves) {
        if (temp[idx-1].length <= 0) {
            continue;
        }
        const x = temp[idx-1].shift()
        if (stack.length > 0 && stack[stack.length-1] == x) {
            stack.pop()
            answer += 2
        } else {
            stack.push(x)
        }
    }

    return answer;
}

File: 프로그래머스/lv1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    answer = 0
    board = list(map(list, zip(*board)))
    result = []
    for m in moves:
        while board[m-1] and board[m-1][0] == 0:
            board[m-1].pop(0)
        if board[m-1]:
            x = board[m-1].pop(0)
            if result and result[-1] == x:
                result.pop()
                answer += 2
            else:
                result.append(x)
    return answer